src/convertToObject.js

```diff
@@ -6,7 +6,26 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
-}
+  let strToObject = sourceString
+    .trim()
+    .split(';')
+    .filter((str) => str.trim());
+
+  strToObject = strToObject.map((str) => {
+    return str
+      .trim()
+      .split(':')
+      .map((s) => s.trim());
+  });
+
+  strToObject = Object.fromEntries(strToObject);
```

Comment:
Instead of reassigning 'strToObject' multiple times, you can chain the methods directly to achieve a cleaner and more readable code structure. Also, consider using 'const' for 'strToObject' since it doesn't need to be reassigned.